application/apps/indexer/addons/file-tools/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+// Idea is to get the file path from the user,
+// read the first chunks from the file and return String
+// 'binary' or 'text' as of now. Later we can add enum for file
+// type.
+
+use anyhow::Result;
+use std::{fs::File, io::Read, path::Path};
+
+const BYTES_TO_READ: usize = 10240;
+
+pub fn detect_file_type(file_path: &Path) -> String {
+    let chunks = fetch_starting_chunk(file_path);
+    let buffer = match chunks {
+        Ok(buffer) => buffer,
+        // TODO: add correct error handling
+        Err(error) => panic!("Error while reading chunks from file: {}", error),
+    };
+    // println!("{:?}", buffer);
+    let is_binary: bool = is_file_binary(&buffer);
+
+    println!("Still in progress... and is likely binary {}", is_binary);
+
+    let result = std::str::from_utf8(&buffer);
+    let content: &str = match result {
+        Ok(file_content) => file_content,
+        Err(err) => panic!(
+            "Error while parsing file using `from_utf8` method from std::str {:?} ",
+            err
+        ),
+    };
+
+    println!("Still in progress.... and content of the file {}", content);
+
+    if is_binary {
+        String::from("The file is likely binary with content\n") + content
+    } else {
+        String::from("The file is not likely binary with content\n") + content
+    }
+}
+
+fn fetch_starting_chunk(file_path: &Path) -> Result<Vec<u8>> {
+    let file = File::open(file_path)?;
+    let mut file = file.take(BYTES_TO_READ.try_into().unwrap());
+    let mut buffer = Vec::new();
```

Comment:
Unwrap is dangerous, it will panic in case of error. Never panic in libs
Also what will happen is `file.size < BYTES_TO_READ`?

Reply:
I will add error handling, and yes I added the code to check the file size and take that into consideration, but somehow when I read entire file, result of the function is wrong and when I go back to `1024` I am getting correct result.